lms/djangoapps/mobile_api/groups/tests.py

```diff
@@ -17,32 +20,112 @@ def setUp(self):
         self.user = UserFactory.create()
         self.client.login(username=self.user.username, password='test')
 
-    def test_get_app_groups(self):
-        url = reverse('get-app-groups')
-        response = self.client.get(url, {'oauth-token' : 'abcd1234'})
-        self.assertEqual(response.status_code, 200)
-        self.assertTrue('groups' in response.data)  # pylint: disable=E1103
-
     def test_create_new_group(self):
-        url = reverse('create-new-group') 
-        response = self.client.post(url, {  'name' : 'TheBestGroup', 
+        # Create new group
+        url = reverse('create-new-group')
+        response = self.client.post(url, {  'name' : 'TheBestGroup',
                                             'description' : 'The group for the best people',
-                                            'privacy' : 'true',
-                                            'admin-id' : '12345',
-                                            'oauth-token' : 'abcd1234'})
+                                            'privacy' : 'open'})
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('id' in response.data)  # pylint: disable=E1103
+        
+        # Delete the groupd just created
+        url = reverse('delete-group', kwargs={'group_id' : response.data['id']}) 
```

Comment:
I would split the delete into another test --- with a non-view generated setup. This test isolation makes it easy to detect flaws immediately